include/realtime_tools/realtime_publisher.hpp

```diff
@@ -210,12 +236,12 @@ class RealtimePublisher
   std::condition_variable updated_cond_;
 #endif
 
-  enum { REALTIME, NON_REALTIME, LOOP_NOT_STARTED };
-  std::atomic<int> turn_;  // Who's turn is it to use msg_?
+  enum class State : int { REALTIME, NON_REALTIME, LOOP_NOT_STARTED };
```

Comment:
You can use a smaller backing type here since there are only 3 cases.

```suggestion
  enum class State : uint8_t { REALTIME, NON_REALTIME, LOOP_NOT_STARTED };
```

Reply:
On the other hand we also don't really profit from using a smaller datatype. In the end it will be filled up to an int on 90% of the systems.